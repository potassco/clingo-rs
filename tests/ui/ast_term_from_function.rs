// ast_term_from_function
use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let args = vec![term1, term2];
    let fun = ast::function(&loc, "name", &args, false).unwrap();
    let term3: ast::Term = fun.into();

    let fun2 = ast::function(&loc, "name2", &args, true).unwrap();
    let term4: ast::Term = fun2.into();

    drop(args);
    let _end = (term3, term4);
}
