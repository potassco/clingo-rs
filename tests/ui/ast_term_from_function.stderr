error[E0505]: cannot move out of `args` because it is borrowed
  --> tests/ui/ast_term_from_function.rs:16:10
   |
9  |     let args = vec![term1, term2];
   |         ---- binding `args` declared here
10 |     let fun = ast::function(&loc, "name", &args, false).unwrap();
   |                                           ----- borrow of `args` occurs here
...
16 |     drop(args);
   |          ^^^^ move out of `args` occurs here
17 |     let _end = (term3, term4);
   |                 ----- borrow later used here
